pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -185,9 +194,18 @@ func (rc *reconciler) reconcile() {
 			}
 			// Check whether timeout has reached the maximum waiting time
 			timeout := elapsedTime > rc.maxWaitForUnmountDuration
-			// Check whether volume is still mounted. Skip detach if it is still mounted unless timeout
-			if attachedVolume.MountedByNode && !timeout {
-				klog.V(5).Infof(attachedVolume.GenerateMsgDetailed("Cannot detach volume because it is still mounted", ""))
+
+			isHealthy, err := rc.nodeIsHealthy(attachedVolume.NodeName)
```

Comment:
merge conflict had to be resolved here to remove the out-of-service node taint logic